Add Emotion recommendations (#73)
guardian/recommendations: emotion.md

> # Using Emotion

Reviewer: ```suggestion
# Using Emotion with React
```

we're specifically setting this is up for use with react right?

> [Emotion](https://emotion.sh/) is a CSS-in-JS library that provides component-scoping, encapsulation and composition for your styles.

Reviewer: ```suggestion
[Emotion](https://emotion.sh/) is a CSS-in-JS library that provides component-scoping, encapsulation and composition for your styles.

It can be used with and without React, but this guide covers our standard use, which is alongside React. In particular it sets out how to use Emotion alongside the React implementation of our design system, [`Source`](https://github.com/guardian/source).
```

## Getting started

```shell
$ yarn add @emotion/react
```

> You need to install `@emotion/react` to use [Source components](https://github.com/guardian/source)

## Configuration

### TypeScript

We recommend using [React's automatic runtime](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html).

In your `tsconfig.json`:

```json
{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "@emotion/react"
  }
}
```

> This tells TypeScript to use React's automatic runtime and lets TypeScript know where it can find the JSX namespace.

### Babel

> You can use the `css` prop, in conjunction with `@emotion/babel-preset-css-prop`:

Reviewer: do you need this if you have the above ts setup?

Author: No, I believe this is handled by the [`jsxImportSource` option](https://emotion.sh/docs/typescript#emotionreact) in your tsconfig's `compilerOptions`:

```
{
  "jsxImportSource": "@emotion/react"
}
```

Author: Should we make it clearer that TypeScript and Babel are 2 different paths for configuration? e.g

```
## Configuration

Emotion may be configured in your `tsconfig.json` or your `.babelrc`, depending on your build setup.

### Approach 1: TypeScript

...

### Approach 2: Babel

...
```

Reviewer: 👍

Reviewer: added a suggestion so will resolve this

```shell
$ yarn add @emotion/babel-preset-css-prop -D
```

In your `.babelrc`:

```json
{
  "presets": ["@emotion/babel-preset-css-prop"]
}
```

> **Note:** If you use `@babel/preset-react` or `@babel/preset-typescript` ensure that `@emotion/babel-preset-css-prop` is inserted _after_ them in your babel config.

### Server side rendering

> If your app is doing server-side rendering, you need to follow the [Advanced Approach](https://emotion.sh/docs/ssr#advanced-approach) to SSR setup. This is because our components use sibling selectors that may be interrupted by the way Emotion embeds `<style>` tags directly into your markup.

Reviewer: ```suggestion
If your app is doing server-side rendering, you need to follow [Emotion's Advanced Approach](https://emotion.sh/docs/ssr#advanced-approach) to SSR setup. This is because our components use sibling selectors that may be interrupted by the way Emotion embeds `<style>` tags directly into your markup.
```

# Using Emotion with React

[Emotion](https://emotion.sh/) is a CSS-in-JS library that provides component-scoping, encapsulation and composition for your styles.

It can be used with and without React, but this guide covers our standard use, which is alongside React. In particular it sets out how to use Emotion alongside the React implementation of our design system, [`Source`](https://github.com/guardian/source).

## Getting started

```shell
$ yarn add @emotion/react
```

> You need to install `@emotion/react` to use [Source components](https://github.com/guardian/source)

> ## Configuration
> 

Reviewer: ```suggestion
## Configuration

Emotion can be configured for typescript or babel, depending on your setup. If you are using both, you only need to do one (TypeScript is simplest).
```

### TypeScript

> We recommend using [React's automatic runtime](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html).

Reviewer: ```suggestion
If you are using TypeScript <= 4.1 and React <=17, it's simplest to use [React's automatic runtime](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) with TypeScript's attendant [jsx](https://www.typescriptlang.org/tsconfig#jsx) and [jsxImportSource}(https://www.typescriptlang.org/tsconfig#jsxImportSource) compiler options.
```
